components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/claims/impl/DefaultClaimHandler.java

```diff
@@ -1100,6 +1107,112 @@ private void handleRoleClaim(AuthenticationContext context, Map<String, String>
         }
     }
 
+    /**
+     * Add the application roles of federated user to remote claims.
+     *
+     * @param stepConfig       StepConfig of current step.
+     * @param context          AuthenticationContext of current authentication flow.
+     * @param idPClaimMappings Claim mappings of IdP of the current step.
+     * @param remoteClaims     Remote claims of the current step authenticated user.
+     * @throws FrameworkException Exception on handling application roles for federated user.
+     */
+    protected void handleApplicationRolesForFederatedUser(StepConfig stepConfig, AuthenticationContext context,
```

Comment:
Can't we use private methods?

Reply:
To add a test to this method, this method needs to be a protected method.
Similarly `handleApplicationRolesForLocalUser` method is also protected method

Reply:
If this method is not intended to be used outside of this class, the access modifier should be `private`.

The testing should be covered from the actual method which invokes the private method

Reply:
This is a newly added test class for this method.
Trying to mock the `handleFederatedClaims` method branches into multiple private methods. This branches to almost all private methods in this class.
Since these cannot be mocked, need to mock the utils classes. There seems to be way too many mocks.
Therefore its better if we can test the `handleApplicationRolesForLocalUser` using it as a protected method.